### ISSUE TITLE
Atlas tech team sprint workflow

### ISSUE THREAD

**Issue by kdembler** [Joystream/atlas#918]
## Purpose

This document describes the process of implementing new features for Atlas as part of sprints. This should help align the team on what the process it and how we follow it. https://github.com/Joystream/atlas/issues/777 is important to fully understand the process

## Sprint organization

Work on new features/improvements will be organized in sprints (milestones), with each sprint having some amount of developers assigned to it and the scope defined as clearly as possible. Each sprint will have a unique Github milestone attached to it so it's easy to see how the progress on specific feature is going. Based on the scope, sprints should also have rough estimates.

While each member of the team may be a part of a different sprint (or possibly couple of those at the same time), it's useful to see the status of the full team, so we will be using a single github board (https://github.com/orgs/Joystream/projects/42) to keep track of all the work. This board will also be shared with the designers, as some sprints may be interdisciplinary and worked across tech/design teams. The board can be filtered to see progress on specific milestone, just the design tasks, etc.

## Sprint workflow

Each tech sprint shall have a unique branch created in the repository. Most of the work for that sprint should be merged into that feature branch. Only when the full feature (sprint) is complete and ready to be released to the community, the feature branch should be merged inside `dev` for staging. Once we're confident the change is working good, `dev` should be merged into `master`. There can be some exceptions to the feature branch policy, for example some app-wide refactor can be targeting `dev` directly. Each PR needs to have at least one approval from another developer on the team. Developers should have write permissions to autonomically merge PRs into their feature branches

## Example sprint

Let's pick an example sprint to see what the workflow should be.

**Sprint: Viewer channel view update**

1. Tech lead with help of designers defines the scope of the sprint and required features
2. Tech lead creates github milestone (`Viewer channel view update`) and branch (`channel-view-update`) corresponding to the sprint
3. Tech lead assigns developers to work on the sprint
4. Teach lead and/or assigned developers create granular issues describing the work to be done and estimate those. All the issues are assigned to the milestone and put on a global Atlas board
5. Assigned developers work on the issues, opening granular PRs to the `channel-view-update` branch
6. Some PRs may be targeting the `dev` branch directly, for example elevating paginated videos grid to a separate components. Those PRs must not break the working `dev` version
7. Each PR must be approved by at least one other team member (preferably someone familiar with the feature being worked on).
8. Once approved, assigned developer can merge the PR into the feature branch.
9. The feature branch may be rebased to incorporate latest changes from `dev`
10. Once the full scope of the sprint is done and working properly on the feature branch, assigned developer will open a PR against `dev`
11. The PR to `dev` needs to be approved and merged by the tech lead.
12. After some staging period on `dev`, PR to `master` should be opened
13. Once PR to `master` is merged and the feature released to the community, the sprint (milestone) can be closed

## Feedback

Any feedback is welcome
